components/dashboard/dashboard.go

```diff
@@ -62,20 +62,12 @@ func (d *Dashboard) GetComponentName() string {
 
 func (d *Dashboard) ReconcileComponent(ctx context.Context,
 	cli client.Client,
-	logger logr.Logger,
+	l logr.Logger,
 	owner metav1.Object,
 	dscispec *dsciv1.DSCInitializationSpec,
 	platform cluster.Platform,
 	currentComponentExist bool,
 ) error {
-	var l logr.Logger
-
-	if platform == cluster.SelfManagedRhods || platform == cluster.ManagedRhods {
```

Comment:
I do support this change but wonder if this was done on purpose so i.e. there is some log scraper/monitor that expect a certain logger name for the dahboard.

@zdtsw @VaishnaviHire may know more ?

Reply:
As I mentioned, as an alternative I can move final logger creation to the component. It even more idiomatic.

Reply:
not i am aware of if any consumer is explicitly checking on a different name between upstream and downstream. 
since we are using different names in dashboard's case,  i kept different names when added  logger

Reply:
and it probably was the reason why the caller was added into each components rather than in a high level (DSC controller). 
it could have done in a different way, to have it in DSC controller with one override only for dashboard if it is running for downstream.

Reply:
It can. But I still do not understand a reason why it should be different from ComponentName only for dashboard.

Reply:
Can we invite dashboard guys here?

Reply:
I checked with dashboard and it looks that it is just legacy and safe to be removed.

Reply:
Thinking a little bit about this, since the dashboard's `GetComponentName() ` always returns the upstream name which then ends up being also the name used for conditions (i.e. condition type is `dashboardReady`), shouldn't the logger name be consistent with that ?

Reply:
yep, that is the point

Reply:
one reason I kept a different name when introduced logger is the label applied on dashboard resources in downstream, e.g pod selector
```
    app: rhods-dashboard
    app.kubernetes.io/part-of: rhods-dashboard
    app.opendatahub.io/rhods-dashboard: 'true'
    deployment: rhods-dashboard
```
so to macth the same log `opendatahub.controllers.DataScienceCluster.DSC.Components.rhods-dashboard `and pod. 
but it might not be a concern at all, when everyone knows `rhods-dashboard` is the same as `dashboard`